File: Setting/Setting/ViewController.m
//
//  ViewController.m
//  Setting
//
//  Created by apple on 2017/5/27.
//  Copyright © 2017年 apple. All rights reserved.
//

#import "ViewController.h"
#import <MessageUI/MessageUI.h>
#import <StoreKit/StoreKit.h>

@interface ViewController ()<SKStoreProductViewControllerDelegate>

@end

@implementation ViewController
- (IBAction)phone:(UIButton *)sender {
    NSString *title = sender.currentTitle;
    
    NSString *urlString = @"";
    
    if ([title isEqualToString:@"打电话"]) {
        // 直接拨打电话
        //NSString *url=[NSString stringWithFormat:@"tel://%@",phoneNumber];
        // 会提示用户是否拨打电话
        //NSString *url=[NSString stringWithFormat:@"telprompt://%@",phoneNumber];
//        urlString = @"tel://15105713500";
         urlString = @"telprompt://15105713500";
    } else if([title isEqualToString:@"发短信"]) {
        urlString = @"sms://15105713500";
    } else if([title isEqualToString:@"发邮件"]) {
        urlString = @"mailto://120903100@qq.com";
    } else if([title isEqualToString:@"上网"]) {
        urlString = @"https://www.baidu.com";
    } else if([title isEqualToString:@"设置"]) {
        urlString = UIApplicationOpenSettingsURLString;
    } else if([title isEqualToString:@"更新"]) {
        urlString = @"https://itunes.apple.com/cn/app/id333206289?mt=8";
    } else if([title isEqualToString:@"评论"]) {
        urlString = @"http://itunes.apple.com/WebObjects/MZStore.woa/wa/viewContentsUserReviews?id=333206289&pageNumber=0&sortOrdering=2&type=Purple+Software&mt=8";;
    } else if([title isEqualToString:@"内部评论"]) {
        [self evaluate];
        return;
    } else if([title isEqualToString:@"QQ"]) {
        urlString = @"mqq://im/chat?chat_type=wpa&uin=120903100&version=1&src_type=web";
//        urlString = @"weixin:";
    } else {
        return;
    }
//    MFMessageComposeViewController
//    MFMailComposeViewController
    NSURL *url = [NSURL URLWithString:urlString];
    if ([[UIApplication sharedApplication] canOpenURL:url]) {
        [[UIApplication sharedApplication] openURL:url];
    }
    
    if ([UIDevice currentDevice].systemVersion.floatValue >= 8.0) {
        
    }
    
}

- (void)evaluate{
    
    //初始化控制器
    SKStoreProductViewController *storeProductViewContorller = [[SKStoreProductViewController alloc] init];
    //设置代理请求为当前控制器本身
    storeProductViewContorller.delegate = self;
    //加载一个新的视图展示
    [storeProductViewContorller loadProductWithParameters:  @{SKStoreProductParameterITunesItemIdentifier : @"333206289"} completionBlock:^(BOOL result, NSError *error) {
      if(!error) {
      }
  }];
    //模态弹出AppStore应用界面
    [self presentViewController:storeProductViewContorller animated:true completion:nil];
}
//https://itunes.apple.com/cn/app/xcode/id497799835
-(void)productViewControllerDidFinish:(SKStoreProductViewController *)viewController{
    [viewController dismissViewControllerAnimated:YES completion:nil];
}

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: Swift练习/Swift iOS应用开发实战/03-设计模式和视图控制器/3.1__搭建计算器的界面/2.1__搭建计算器的界面/ViewController.swift
//
//  ViewController.swift
//  2.1__搭建计算器的界面
//
//  Created by M-coppco on 16/6/19.
//  Copyright © 2016年 M-coppco. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    @IBOutlet weak var resultL: UILabel!
    var firstNumber : Double = 0.0  //第一个数字
    var secondNumber : Double = 0.0  //第二个数字
    var decimalPointFlag : Bool = false   //小数点标记
    var isSecondNumber : Bool = false  //是否输入了第二个数
    var operatorFlag : String = ""  //操作符
    var isResult : Bool = false

    override func viewDidLoad() {
        super.viewDidLoad()
        resultL.adjustsFontSizeToFitWidth = true //自适应
        self.original()
    }

    //恢复初始状态
    func original() {
        firstNumber = 0.0
        secondNumber = 0.0
        isSecondNumber = false
        decimalPointFlag = false
        operatorFlag = ""
        resultL.text = "0"
        isResult = false
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    //数字按钮点击
    @IBAction func numberTap(sender: UIButton) {
        
        if isResult {
            self.original()
        }
        
        //清空数字
        if resultL.text == "0" || (isSecondNumber && secondNumber == 0.0) {
            self.resultL.text = ""
        }
        //显示文本
        resultL.text = resultL.text!  + sender.currentTitle!
        if isSecondNumber {
            secondNumber = (resultL.text! as NSString).doubleValue
        } else {
            firstNumber = (resultL.text! as NSString).doubleValue
        }
        print(firstNumber, secondNumber)
    }
    //小数点
    @IBAction func decimalPoint(sender: UIButton) {
        if isResult {
            self.original()
        }
        
        //最开始状态decimalPointFlag为false, 那么会进入方法中,在结果中添加小数点,然后decimalPointFlag状态为true, 以后点击. 都不会再拼接点了
        if !decimalPointFlag {
            resultL.text = resultL.text! + "."
            
            if isSecondNumber {
                secondNumber = (resultL.text! as NSString).doubleValue
            } else {
                firstNumber = (resultL.text! as NSString).doubleValue
            }
            
            decimalPointFlag = !decimalPointFlag
        }
    }
    
    //加减乘除
    @IBAction func operationTap(sender: UIButton) {
        if firstNumber != 0.0 {
            isSecondNumber = true
            decimalPointFlag = false
            switch sender.currentTitle! {
            case "+":
                operatorFlag = "+"
            case "-":
                operatorFlag = "-"
            case "*":
                operatorFlag = "*"
            case "/":
                operatorFlag = "/"
            default :
                operatorFlag = ""
            }
        }
    }
    //等于=
    @IBAction func resultTap(sender: UIButton) {
        if isSecondNumber {
            //除数不能为0
            if operatorFlag == "/" && secondNumber == 0.0 {
                print("除数不能为0")
                self.original()
                return
            }
            var resultNumber : Double = 0.0
            switch operatorFlag {
            case "+":
                resultNumber = firstNumber + secondNumber
            case "-":
                resultNumber = firstNumber - secondNumber
            case "*":
                resultNumber = firstNumber * secondNumber
            case "/":
               resultNumber = firstNumber / secondNumber
            default :
                resultNumber = 0.0
            }
            resultL.text = resultNumber.description
            isResult = true
        }
    }
    
    //删除
    @IBAction func deleteTap(sender: UIButton) {
        self.original()
    }
    //百分号
    @IBAction func baifen(sender: UIButton) {
    }
    //+/-
    @IBAction func revert(sender: UIButton) {
        if !isSecondNumber && firstNumber != 0.0{
            firstNumber = -firstNumber
            resultL.text = firstNumber.description
        } else if isSecondNumber && secondNumber != 0.0{
            secondNumber = -secondNumber
            resultL.text = secondNumber.description
        } else {
            return
        }

    }
    
}

